common/txmgr/types/tx.go

```diff
@@ -109,3 +112,192 @@ type TxMeta[ADDR types.Hashable, TX_HASH types.Hashable] struct {
 	// SeqNumbers is used by CCIP for tx to committed sequence numbers correlation in logs
 	SeqNumbers []uint64 `json:"SeqNumbers,omitempty"`
 }
+
+type TxAttempt[
+	CHAIN_ID ID,
+	ADDR types.Hashable,
+	TX_HASH, BLOCK_HASH types.Hashable,
+	R ChainReceipt[TX_HASH],
+	FEE Fee,
+	ADD any,
+] struct {
+	ID    int64
+	TxID  int64
+	Tx    Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD]
+	TxFee FEE
+	// ChainSpecificFeeLimit on the TxAttempt is always the same as the on-chain encoded value for fee limit
+	ChainSpecificFeeLimit   uint32
+	SignedRawTx             []byte
+	Hash                    TX_HASH
+	CreatedAt               time.Time
+	BroadcastBeforeBlockNum *int64
+	State                   TxAttemptState
+	Receipts                []Receipt[R, TX_HASH, BLOCK_HASH] `json:"-"`
+	TxType                  int
+}
+
+func (a TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD]) String() string {
+	return fmt.Sprintf("TxAttempt(ID:%d,TxID:%d,Fee:%s,TxType:%d", a.ID, a.TxID, a.TxFee, a.TxType)
+}
+
+func (a TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD]) Fee() FEE {
```

Comment:
Pointer receivers should be more efficient for `Tx` and `TxAttempt`:
```suggestion
func (a *TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD]) Fee() FEE {
```